base/compiler/typeinfer.jl

```diff
@@ -617,6 +620,9 @@ function typeinf_ext(interp::AbstractInterpreter, mi::MethodInstance)
             end
         end
     end
+    if ccall(:jl_get_module_infer, Cint, (Any,), method.module) == 0
+        return retrieve_code_info(mi)
+    end
```

Comment:
same here. and also because this code is holding a lock, which must be released before returning

Reply:
I am not sure this code is needed here, though, or if it should go in `jl_generate_fptr_impl` with the other heuristics. It feels like the implementation here is instead dictating policy decisions here, and ignoring the user's specific instruction.